### PR TITLE
Fix inherited icons for script classes

### LINKED ISSUE #24657: When a script extends from a class with a custom icon, the base class icon is used instead
My expectation would be that when extending from a class, it would show the class icon, not the base class icon.
In a similar fashion, when extending from a class with a custom icon, and naming the new class with class_name and providing a path to new icon, no icon is used at all - I would expect the new icon path to be used in that case (always to 'topmost' one).

Usecase:
shot_181229_163633

I use custom classes for Effects and EffectComponents. EffectComponents are usually extended, with some added functionality - I then lose the benefits of custom icons (which, when they work, are awesome : )

@willnationsdev

### PR DESCRIPTION
**Pull request by willnationsdev**
This should close #24657. I added a loop that goes through the base classes in the script class and checks if any of them have icons, all the way up until there aren't any more scripts.

### CODE REVIEW

File: editor/editor_node.cpp [hunk 1]
```diff
@@ -3242,9 +3242,23 @@ Ref<Texture> EditorNode::get_class_icon(const String &p_class, const String &p_f
 		RES icon;
 		if (FileAccess::exists(icon_path)) {
 			icon = ResourceLoader::load(icon_path);
-		}
-		if (!icon.is_valid()) {
-			icon = gui_base->get_icon(ScriptServer::get_global_class_base(p_class), "EditorIcons");
+			if (icon.is_valid()) {
```
Comment (akien-mga) [inline editor/editor_node.cpp]: This changes the previous logic, I guess you wanted `!icon.is_valid()`?

Comment (akien-mga) [reply editor/editor_node.cpp]: Ping.

Comment (willnationsdev) [reply editor/editor_node.cpp]: I've amended the implementation. Recompiling on my end. Will upload again when next available.

**Comment by willnationsdev** [comment]
Updated

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.1.

**Pull request merged**